### PR TITLE
Implemented __cnfline using bitarray

### PR DESCRIPTION
**Pull request by t0mpr1c3**
NB installation of bitarray requires python3.7-dev

### CODE REVIEW

**Comment by t0mpr1c3** [comment]
Closed PR due to dirty git history, will reopen after rebase.

File: src/main/python/ayab/plugins/ayab_plugin/ayab_control.py [hunk 1]
```diff
@@ -401,28 +386,28 @@ def __cnfLine(self, lineNumber):
...
+                    if 0 <= pxlNumber and pxlNumber < MACHINE_WIDTH:
+                        bits[pxlNumber] = True
 
             # TODO implement CRC8
             crc8 = 0x00
 
             # send line to machine
             if self.__infRepeat:
-                self.__ayabCom.cnf_line(reqestedLine, bytes, 0, crc8)
+                self.__ayabCom.cnf_line(reqestedLine, bits.tobytes(), 0, crc8)
```
Comment (chris007de) [inline src/main/python/ayab/plugins/ayab_plugin/ayab_control.py:408]: Have you verified that the bits.tobytes() operation gives expected result (i.e. have you knitted successfully?)

Comment (t0mpr1c3) [reply src/main/python/ayab/plugins/ayab_plugin/ayab_control.py:408]: Only on toy examples so far, but I'll get round to it.

Comment (t0mpr1c3) [reply src/main/python/ayab/plugins/ayab_plugin/ayab_control.py:408]: It needed a change to ensure an integer index. The ordering of bits is correct.

**Comment by chris007de** [comment]
Is this ready to merge?

**Comment by t0mpr1c3** [comment]
Yes, it's good to go.

**Pull request merged**